### PR TITLE
UCT/IB/DC: add env for reverse_sl (default: same value of sl)

### PR DESCRIPTION
**Pull request by roiedanino**
## What
Set reverse_sl to the same value of sl (service level) configured by `UCX_IB_SL` env.

## Why ?
Currently on certain device configurations reverse_sl is 0 if not set.

### CODE REVIEW

File: src/uct/ib/dc/dc_mlx5_devx.c [hunk 1]
```diff
@@ -184,7 +184,8 @@ ucs_status_t uct_dc_mlx5_iface_devx_set_srq_dc_params(uct_dc_mlx5_iface_t *iface
     }
     UCT_IB_MLX5DV_SET(set_xrq_dc_params_entry_in, in, mtu, iface->super.super.super.config.path_mtu);
     UCT_IB_MLX5DV_SET(set_xrq_dc_params_entry_in, in, sl, iface->super.super.super.config.sl);
-    UCT_IB_MLX5DV_SET(set_xrq_dc_params_entry_in, in, reverse_sl, iface->super.super.super.config.sl);
+    UCT_IB_MLX5DV_SET(set_xrq_dc_params_entry_in, in, reverse_sl,
+                      iface->super.super.super.config.reverse_sl);
     UCT_IB_MLX5DV_SET(set_xrq_dc_params_entry_in, in, cnak_reverse_sl, iface->super.super.super.config.sl);
```
Comment (Artemy-Mellanox) [inline src/uct/ib/dc/dc_mlx5_devx.c:189]: probably set this value to cnak_reverse_sl too?

**Review by Artemy-Mellanox — APPROVED** [review]

File: src/uct/ib/dc/dc_mlx5_ep.c [hunk 2]
```diff
@@ -36,6 +39,11 @@ uct_dc_mlx5_set_dgram_seg(uct_ib_mlx5_txwq_t *txwq,
     to_av->fl_mlid      = iface->tx.av_fl_mlid;
     to_av->rlid         = av->rlid;
 
+    /* Setting reverse_sl */
+    to_av->dqp_dct &= ~(REVERSE_SL_MASK);
+    to_av->dqp_dct |= (iface->super.super.super.config.reverse_sl &
+                       REVERSE_SL_MASK);
```
Comment (yosefe) [inline src/uct/ib/dc/dc_mlx5_ep.c:45]: extra ( ) in both lines

File: src/uct/ib/rc/accel/rc_mlx5_iface.c [hunk 3]
```diff
@@ -826,6 +826,11 @@ UCS_CLASS_INIT_FUNC(uct_rc_mlx5_iface_common_t, uct_iface_ops_t *tl_ops,
     status = uct_ib_mlx5_iface_select_sl(&self->super.super,
                                          &mlx5_config->super,
                                          &rc_config->super);
+    self->super.super.config.reverse_sl = (rc_config->super.reverse_sl ==
+                                           UCS_ULUNITS_AUTO) ?
+                                                  self->super.super.config.sl :
+                                                  rc_config->super.reverse_sl;
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_iface.c:832]: seems like code duplication with uct_ib_iface_config_select_reverse_sl

File: src/uct/ib/ud/accel/ud_mlx5.c [hunk 4]
```diff
@@ -998,6 +998,10 @@ static UCS_CLASS_INIT_FUNC(uct_ud_mlx5_iface_t, uct_md_h tl_md,
     if (status != UCS_OK) {
         return status;
     }
+    self->super.super.config.reverse_sl =
+            (config->super.super.reverse_sl == UCS_ULUNITS_AUTO) ?
+                    self->super.super.config.sl :
+                    config->super.super.reverse_sl;
```
Comment (yosefe) [inline src/uct/ib/ud/accel/ud_mlx5.c:1004]: seems like code duplication with [rc_mlx5_iface.c](https://github.com/openucx/ucx/pull/9489/files#diff-7ab6de1abc368d4bd4c2c32497d4cf4c02142f9239d3df36c9f621b7bbed2ade)

File: src/uct/ib/dc/dc_mlx5_ep.c (hunk 2 above)
Comment (roiedanino) [reply src/uct/ib/dc/dc_mlx5_ep.c:45]: Fixing in: https://github.com/openucx/ucx/pull/9510

**Pull request merged**